src/test/java/org/apache/commons/io/build/URIOriginTest.java

```diff
@@ -35,4 +39,21 @@ public void beforeEach() {
         setOriginRw(new URIOrigin(Paths.get(FILE_NAME_RW).toUri()));
     }
 
+    @Test
+    void testGetInputStreamHttpURI() throws Exception {
+        final String uri = "https://example.com";
+        final AbstractOrigin.URIOrigin origin = new AbstractOrigin.URIOrigin(new URI(uri));
+        try (final InputStream in = origin.getInputStream()) {
+            assertNotEquals(-1, in.read());
+        }
+    }
+
+    @Test
+    void testGetInputStreamHttps() throws Exception {
+        final String uri = "https://example.com";
+        final AbstractOrigin.URIOrigin origin = new AbstractOrigin.URIOrigin(new URI(uri));
+        try (final InputStream in = origin.getInputStream()) {
+            assertNotEquals(-1, in.read());
+        }
+    }
```

Comment:
Add tests for 'http' just like you have 'https'. Use JUnit parameterized tests to reduce code duplication.

Reply:
Yes, updated